src/lib.rs

```diff
@@ -308,6 +309,16 @@ impl Core {
                 test_println!("channel full");
                 return Err(TrySendError::Full(()));
             }
+            // if the current state is marked as data written by the current (or even future)
+            // generation we can spin as probably our tail is behind the actual tail
+            // but if the state is marked as data written by the previous generation and nobody
+            // is reading it then we don't have any available slots
```

Comment:
some minor grammar nits:
```suggestion
            // if the current state is marked as data written by the current (or even future)
            // generation, we can spin, as our tail is probably behind the actual tail.
            // however, if the state is marked as data written by the previous generation and nobody
            // is reading it, then we don't have any available slots.
```